analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -282,6 +279,19 @@ private class Graph(private val nodeMap: MutableMap<Identifier, Set<Identifier>>
         Graph(nodeMap.filter { it.key in subNodes }
             .mapValuesTo(mutableMapOf()) { e -> e.value.filterTo(mutableSetOf()) { it in subNodes } })
 
+    /**
+     * Search for the single package that represents the main project. This is the only package without a version.
+     * Fail if no single package with this criterion can be found.
+     */
+    fun projectId(): Identifier =
+        nodes().filter { it.version.isBlank() }.let { idsWithoutVersion ->
+            require(idsWithoutVersion.size == 1) {
+                "Expected exactly one unique package without version but got ${idsWithoutVersion.joinToString()}."
+            }
+
+            idsWithoutVersion.first()
```

Comment:
Just out of curiosity, are you deliberately not using `idsWithoutVersion.single()` here because you want to customize the exception message?

Reply:
This was the original way this check was implemented before my changes. So I just restored it.